meson.build

```diff
@@ -0,0 +1,119 @@
+project(
+  'libdicom',
+  'c',
+  license: 'MIT',
+  meson_version : '>=0.54',
+  version: '0.1.0',
+)
+
+# dependencies
+system_uthash = get_option('system_uthash')
+if system_uthash.enabled() or system_uthash.auto()
+  cc = meson.get_compiler('c')
+  found_uthash = cc.check_header(
+    'utarray.h',
+    required : system_uthash
+  )
+  found_uthash = found_uthash and cc.check_header(
+    'uthash.h',
+    required : system_uthash
+  )
+  if found_uthash
+    if system_uthash.auto()
+      message('Using system uthash')
+    endif
+    uthash = ''
+  else
+    message('Using bundled uthash')
+    uthash = 'uthash'
+  endif
+else
+  uthash = 'uthash'
+endif
+check = dependency(
+  'check',
+  required : false,
+  version : '>=0.9.6',
+)
+
+# options
+cfg = configuration_data()
+suffixed_version = meson.project_version()
+version_suffix = get_option('version_suffix')
+if version_suffix != ''
+  suffixed_version += '-' + version_suffix
+endif
+cfg.set(
+  'HAVE_CHECK',
+  check.found(),
+  description : 'Define if the Check package is available.',
+)
+cfg.set_quoted(
+  'SRCDIR',
+  meson.current_source_dir().replace('\\', '\\\\'),
+  description : 'Define to the path to the source directory.',
+)
+cfg.set_quoted(
+  'SUFFIXED_VERSION',
+  suffixed_version,
+  description : 'The package version string including any suffix.',
+)
+configure_file(
+  output : 'config.h',
+  configuration : cfg,
+)
+
+# libdicom
+lib_sources = [
+  'src/dicom.c',
+  'src/dicom-data.c',
+  'src/dicom-dict.c',
+  'src/dicom-file.c',
+]
+install_headers(
+  'src/dicom.h',
+  subdir : 'dicom',
```

Comment:
How about `dcm`? The C API uses `dcm_` and `DCM_` as the namespace prefix, perhaps we should use it for the include prefix as well?

Reply:
I used the same prefix that Autotools had.  I'd be inclined to leave it as is, though, since the library is `libdicom` and not `libdcm`.